### PR TITLE
Implement mostly working version of RF.

### PR DESCRIPTION
**Pull request by gpestre**
The algorithm is in place, but it's not actually random yet (because it always uses a seed of -1 for the bootstrapping).

Below I proposed a `SeedGenerator` object that is initialized within each RandomForest instance, and provides a stream of random seeds that can be used for (a) bootstrapping and (b) to pass to each tree, which has its own SeedGenerator to select different random subsets of features at each split.

Hope it's not overkill, but seemed like a straighforward way to allow replicability for our debugging, without ending up with the same bootstrap each time. (See more in issue #65 ).

### CODE REVIEW

File: src/datasets.cpp [hunk 1]
```diff
@@ -741,4 +740,37 @@ DataLoader::DataLoader(std::string filename)
...
+    /**
+     * Obtain a new non-negative integer seed
+     * (returns -1 if SeedGenerator is in non-deterministic mode).
+     */
+    if (this->meta_seed_==-1) {
+        return -1;
+    } else {
+        return -1;  // PLACEHOLDER.
+    }
+}
```
Comment (gpestre) [inline src/datasets.cpp:768]: @johannes-kk any suggestions on how to implement this for #65 ?
We could also have a `.shuffled_indices(nfeatures,mtry)` method in this object; or let DecisionTree handle that shuffling on its own, given an integer seed from the this generator.

File: src/random_forest.cpp [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            bootstrap, this->regression_, this->loss_, this->mtry_,
+            this->max_height_, this->max_leaves_, this->min_obs_, this->max_prop_, tree_seed
+        );
+        tree.fit();
+        this->trees_.push_back(tree);
+    }
+    this->fitted_ = true;
+}
+
+DataVector RandomForest::predict(DataFrame* testdata) const
```
Comment (johannes-kk) [inline src/random_forest.cpp:133]: All the from-scratch Pandas functionality really comes in handy here 😄

File: src/datasets.cpp (hunk 1 above)
Comment (johannes-kk) [reply src/datasets.cpp:768]: Good question. I'll have to think about that. Will approve and merge this PR, but we have the separate #65 to track that. Probably a bit over-engineered like you suggested, but the replicability is a nice touch to be able to really make apples-to-apples comparisons of parallelised execution time.

**Review by johannes-kk — APPROVED** [review]
lgtm
